.github/workflows/update-docs.yml

```diff
@@ -0,0 +1,44 @@
+name: Update Docs
+
+on:
+  push:
+    branches:
+        - main
```

Comment:
Do we really need to regenerate doc for every push to main? Should it be done only for tags? I understand that the PR is submitted only if some changes are detected, so I guess this would still be fine. Maybe then should you "pin" the oxygen version so that this limits the changes simply due to the doxygen version change?

Reply:
Right, I thought about it but it'd be nice to use the newer doxygen - I think it's just a matter of approving the PR when it happens, and then we go back to quiet. I was a little concerned that there would be timestamps or other "random" changes, but so far so good.

How about if it becomes an issue, I pin it?